search-service/src/test/kotlin/com/egm/stellio/search/csr/handler/ContextSourceRegistrationHandlerTests.kt

```diff
@@ -0,0 +1,198 @@
+package com.egm.stellio.search.csr.handler
+
+import arrow.core.left
+import arrow.core.right
+import com.egm.stellio.search.common.config.SearchProperties
+import com.egm.stellio.search.csr.model.ContextSourceRegistration
+import com.egm.stellio.search.csr.service.ContextSourceRegistrationService
+import com.egm.stellio.shared.config.ApplicationProperties
+import com.egm.stellio.shared.model.AlreadyExistsException
+import com.egm.stellio.shared.model.ResourceNotFoundException
+import com.egm.stellio.shared.util.*
+import com.ninjasquad.springmockk.MockkBean
+import io.mockk.coEvery
+import io.mockk.coVerify
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.context.properties.EnableConfigurationProperties
+import org.springframework.boot.test.autoconfigure.web.reactive.AutoConfigureWebTestClient
+import org.springframework.boot.test.autoconfigure.web.reactive.WebFluxTest
+import org.springframework.core.io.ClassPathResource
+import org.springframework.http.HttpHeaders
+import org.springframework.http.MediaType
+import org.springframework.security.test.web.reactive.server.SecurityMockServerConfigurers.csrf
+import org.springframework.security.test.web.reactive.server.SecurityMockServerConfigurers.mockJwt
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.web.reactive.server.WebTestClient
+
+@ActiveProfiles("test")
+@EnableConfigurationProperties(ApplicationProperties::class, SearchProperties::class)
+@AutoConfigureWebTestClient(timeout = "30000")
+@WebFluxTest(ContextSourceRegistrationHandler::class)
+class ContextSourceRegistrationHandlerTests {
+
+    @MockkBean
+    private lateinit var contextSourceRegistrationService: ContextSourceRegistrationService
+
+    @Autowired
+    private lateinit var webClient: WebTestClient
+
+    private val csrUri = "/ngsi-ld/v1/csourceRegistrations"
+    private val id = "urn:ngsi-ld:ContextSourceRegistration:1".toUri()
+    private val endpoint = "http://my.csr.endpoint/".toUri()
+
+    @BeforeAll
+    fun configureWebClientDefaults() {
+        webClient = webClient.mutate()
+            .apply(mockJwt().jwt { it.subject(MOCK_USER_SUB) })
+            .apply(csrf())
+            .defaultHeaders {
+                it.accept = listOf(JSON_LD_MEDIA_TYPE)
+                it.contentType = JSON_LD_MEDIA_TYPE
+            }
+            .build()
+    }
+
+    @Test
+    fun `get contextSourceRegistration by id should return 200 when contextSourceRegistration exists`() =
```

Comment:
it is allowed to use the `CSR` acronym to make things shorter (whille still readable) ;)